### DISCUSSION TITLE
Infinite scrolling with a seamless loop to the start

### DISCUSSION THREAD

**Discussion by tinokaartovuori** [idiotWu/smooth-scrollbar#515] [Q&A]
Hello!

I am currently working on creating a website similar to [Jantana Hennard's Portfolio](https://jant.fr/) and I am interested in replicating the infinite scrolling feature. From my research, it appears that the developer is using `smooth-scrollbar` on the website. I have also noticed that the `translate3d` property is applied separately to each child container rather than the entire scroll container.

I am wondering if there is a specific plugin or implementation that can be used for this type of feature. Can you provide some guidance on how I can approach the implementation of this feature on my website?

Thank you!

**Comment by sadeghbarati** [comment]
Wow, I never get tired of scrolling 💯 

Sorry, I'm not that Pro to help you, but from what I saw

- Uses `smooth-scrollbar` per page, not in the layout (app-wide)
- Lock `smooth-scrollbar` x, y scrolling and only get delta on `wheel` and `touch` events, [simple demo](https://stackblitz.com/edit/vitejs-vite-1xkun5?file=src/App.vue)
- Uses gsap [`ticker`](https://greensock.com/docs/v3/GSAP/gsap.ticker) and calculates mathematic stuffs? xd

**Reply by tinokaartovuori** [reply]
Thank you for the valuable information!

>  _"Uses smooth-scrollbar per page, not in the layout (app-wide)"_

Yea, this is something that I noticed too! This is smart since if you can change the behaviour between pages. In fact in the portfolio I provided, this infinite scroll is only used for the home page. 👍

> _"Lock smooth-scrollbar x, y scrolling and only get delta on wheel and touch events, [simple demo](https://stackblitz.com/edit/vitejs-vite-1xkun5?file=src/App.vue)_"

This is actually awesome. I tried to do this in a different way for hours and did not succeed XD. Luckily you had the solution for this. I am also wondering that is there an easy way to find other plugins for `smooth-scrollbar`? I only found the `overscroll` one pretty much. I tried this solution you provided and it works like a charm! 🔥

> _"Uses gsap [ticker](https://greensock.com/docs/v3/GSAP/gsap.ticker) and calculates mathematic stuffs? xd"_

This one actually sounds a little scary since I am not familiar with gsap ticker at all but since now I have the delta on `wheel` and `touch` I can maybe figure something out. **How did you notice that gsap ticker is used here?**

Thank you for your help. 👍 This was actually very useful information (even if you claim not to be Pro)

**Reply by sadeghbarati** [reply]
> How did you notice that gsap ticker is used here?

Developer Tools > Sources/Debugger/Network

I wished [he](https://github.com/patrickheng) turned sourcemap on, so we simply understood what's going on 😄 instead of diggin' into compiled codes 🤯 


Looks like you using Nuxt for this project, we worked on a simple [Vue wrapper](https://www.npmjs.com/package/@smooth-scrollbar-contrib/vue-test) It would be good if you test it out Thanks ❤️

**Reply by tinokaartovuori** [reply]
I might give the Vue wrapper a try! :) Tried to use the `gsap ticker` and it wasn't scary at all!
I have still plenty of stuff to figure out bit this is a nice start!

Thank you! 🔥

**Reply by sadeghbarati** [reply]
I'm glad I could help you a little 

Good Luck!
